src/UpdatedPackagesCollector.js

```diff
@@ -127,7 +127,19 @@ export default class UpdatedPackagesCollector {
 
   collectUpdates() {
     return this.packages.filter(pkg => {
-      return this.updatedPackages[pkg.name] || (this.flags.onlyExplicitUpdates ? false : this.dependents[pkg.name]) || this.flags.canary;
+      if (this.updatedPackages[pkg.name]) {
+        return true;
+      }
+
+      if (!this.flags.onlyExplicitUpdates && this.dependents[pkg.name]) {
+        return true;
+      }
+
+      if (this.flags.canary && !pkg.isPrivate()) {
+        return true;
```

Comment:
Should we just return false if it's private or are we allowing private in the other cases?